### PR TITLE
[*] Command Login - Fix login to match old behaviour

### PR DESCRIPTION
**Pull request by VincentMolinie**
As you mentioned here: https://github.com/ForestAdmin/lumber/pull/295, the login has regressed. The purpose of this PR is to implement the same behaviour as before and adding the new parameters

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,10 @@
 - Commands - Remove command run and install.
 - Command Generate - Install Forest Admin middleware.
 
+### Fixed
+- Command Login - Fix login only enable connection (And not account creation).
+- Command Login - Verify the validity of the token passed in paramaters.
```
Comment (VincentMolinie) [inline CHANGELOG.md]: If you have better ideas on the changelog, feel free

Comment (rap2hpoutre) [reply CHANGELOG.md]:
```suggestion
- Command Login - Check token validity before login.
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -9,6 +9,10 @@
 - Commands - Remove command run and install.
 - Command Generate - Install Forest Admin middleware.
 
+### Fixed
+- Command Login - Fix login only enable connection (And not account creation).
```
Comment (rap2hpoutre) [inline CHANGELOG.md]:
```suggestion
- Command Login - Disable account creation.
```

File: services/authenticator.js [hunk 3]
```diff
@@ -12,9 +12,29 @@ const logger = require('./logger');
...
 function Authenticator() {
+  this.pathToLumberrc = `${os.homedir()}/.lumberrc`;
+
+  this.saveToken = token => fs.writeFileSync(this.pathToLumberrc, token);
+
+  this.isTokenCorrect = (email, token) => {
+    const sessionInfo = parseJwt(token);
+    if (sessionInfo) {
+      if ((sessionInfo.exp * 1000) <= Date.now()) {
+        logger.warn('You tried to use a token an expired token');
```
Comment (rap2hpoutre) [inline services/authenticator.js]:
```suggestion
        logger.warn('Your token has expired.');
```

File: services/authenticator.js [hunk 4]
```diff
@@ -12,9 +12,29 @@ const logger = require('./logger');
...
+        return false;
+      }
+
+      if (sessionInfo.data.data.attributes.email === email) {
+        return true;
+      }
+      logger.warn('You tried to use a token of another account');
```
Comment (rap2hpoutre) [inline services/authenticator.js]:
```suggestion
      logger.warn('Your token is invalid.');
```

I think we should stay vague in this catch-all case. BTW, I'm not sure this message is always true: a user that as a broken session data may be in that case too.

> _User type "p@ssw0rd"..._
> You tried to use a password of another account
> _Wohoo, I stole something and the system helps me understand how to do the next step!_

File: utils/authenticator-helper.js [hunk 5]
```diff
@@ -12,7 +13,7 @@ function parseJwt(token) {
 
     return JSON.parse(jsonPayload);
   } catch (error) {
-    console.error('Session token parsing unexpected error:', error);
+    logger.error('Your session token seems to be wrong');
```
Comment (rap2hpoutre) [inline utils/authenticator-helper.js]:
```suggestion
    logger.error('Your session token is invalid.');
```

File: services/authenticator.js (hunk 4 above)
Comment (VincentMolinie) [reply services/authenticator.js]: Yeah but you need to have the token 🤔. You can't find it magically

**Comment by VincentMolinie** [comment]
> dmin authentication token: [input is hidden]
> >> Invalid token. Please enter your authentica

- I added a validation for emails like the one on the server ;)
- Hmm and `CTRL + C` for me is a normal behaviour when trying to exit. Furthermore, there is a copy button so it should be pretty easy to copy

**Comment by VincentMolinie** [comment]
FOr now I would leave the code like that. The incoming refacto of the prompter should easily refactorize these line of code 😉

**Pull request merged**